src/main/java/mate/academy/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,55 @@
+package mate.academy.repository.impl;
+
+import java.util.List;
+import mate.academy.model.Book;
+import mate.academy.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```

Comment:
If you use Lombok, you can add @RequiredArgsConstructor above class and don`t need this constructor and @autowired